src/protocol.py

```diff
@@ -90,19 +104,30 @@ def isBitSetWithinBitfield(fourByteString, n):
     x, = unpack('>L', fourByteString)
     return x & 2**n != 0
 
-
 # ip addresses
 
 
-def encodeHost(host):
-    """Encode a given host to be used in low-level socket operations"""
-    if host.find('.onion') > -1:
-        return '\xfd\x87\xd8\x7e\xeb\x43' + base64.b32decode(
-            host.split(".")[0], True)
-    elif host.find(':') == -1:
-        return '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xFF\xFF' + \
-            socket.inet_aton(host)
-    return socket.inet_pton(socket.AF_INET6, host)
+if PY3:
```

Comment:
Here I would refactor the original code. Create new constants for the raw strings ,and then you can keep the old codepath inside the function. Like this:

```
if sx.PY2:
    ONION_PREFIX = '\xfd\x87\xd8\x7e\xeb\x43'
else:
    ONION_PREFIX = b'\xfd\x87\xd8\x7e\xeb\x43'
```

Reply:
python2.7 has no problem with bytes literals